src/lib/sci/index.ts

```diff
@@ -1,169 +1,136 @@
-        sci_timed = sci_secs * 60 * 60 * 24 * 7;
-      } else if (months.includes(this.functionalUnitTimeUnit)) {
-        if (this.functionalUnitTimeUnit === 'm') {
-          console.warn(
-            'you are using `m` as your time unit. This is interpreted as months. If you meant minutes, please use `min` instead.'
-          );
-        }
-        sci_timed = sci_secs * 60 * 60 * 24 * 7 * 4;
-      } else if (years.includes(this.functionalUnitTimeUnit)) {
-        sci_timed = sci_secs * 60 * 60 * 24 * 365;
-      } else {
-        throw new Error(
-          'functional-unit-time is not in recognized unit of time'
-        );
-      }
-
-      // apply any functional unit conversion to time-corrected SCI score
-      let factor = 1; // default functional unit is 1 - i.e. do not change current value
-      this.functionalUnit = input['functional-unit'];
-      if (
-        this.functionalUnit in input &&
-        input[this.functionalUnit] !== 'none' &&
-        input[this.functionalUnit] !== ''
-      ) {
-        factor = input[this.functionalUnit];
-      }
-
-      /*
-       * Sci currently in whole single units of time - multiply by duration to convert to user-defined span of time.
-       */
-      sci_timed_duration = sci_timed * this.functionalUnitTimeValue;
-      input['sci'] = sci_timed_duration / factor;
+  functionalUnitTime = {unit: '', value: 1};
+  errorBuilder = buildErrorMessage(this.constructor);
+
+  /**
+   * Configures the SCI Plugin.
+   */
+  public async configure(): Promise<ModelPluginInterface> {
+    return this;
+  }
+
+  /**
+   * Calculate the total emissions for a list of inputs.
+   */
+  public async execute(inputs: ModelParams[]): Promise<any[]> {
+    const tunedInputs = inputs.map(input => {
+      const safeInput = this.validateInput(input);
+      input = Object.assign(input, safeInput);
+
+      this.parseTime(input);
+
+      const sciSecs = this.calculateSciSeconds(input);
+      const sciTimed = this.convertSciToTimeUnit(sciSecs);
+      const factor = this.getFunctionalUnitConversionFactor(input);
+      const sciTimedDuration = sciTimed * this.functionalUnitTime.value;
+
+      input['carbon'] = input['carbon'] ?? sciSecs;
+      input['sci'] = sciTimedDuration / factor;
 
       return input;
     });
 
-    return tunedinputs;
+    return tunedInputs;
   }
 
-  async configure(
-    staticParams: object | undefined
-  ): Promise<ModelPluginInterface> {
-    this.staticParams = staticParams;
-    return this;
+  /**
+   * Gets the conversion factor based on the functional unit specified in the input.
+   * If the 'functional-unit' exists in the input and is not 'none' or an empty string,
+   * returns the value; otherwise, defaults to 1.
+   */
+  private getFunctionalUnitConversionFactor(input: ModelParams): number {
+    const functionalUnit = input['functional-unit'];
+
+    return functionalUnit in input &&
+      input[functionalUnit] !== 'none' &&
+      input[functionalUnit] !== ''
```

Comment:
input[functionalUnit] !== '' instead of this we can check like !(input[functionalUnit])

Reply:
it is not the same, with `!(input[functionalUnit])`  checks its not undefined, but with `input[functionalUnit] !== ''` checks if the variable exists and not an empty string